src/test/xmeta/test_xmeta_emit.cpp

```diff
@@ -283,16 +283,427 @@ std::vector<uint8_t> run_and_save_to_memory(std::istringstream & test_idl, std::
     return reader.save_to_memory();
 }
 
-template<typename T>
-auto find_type_by_name(table<T> const& type, std::string name, std::string namespace_name)
+void VerifyTypeRefName(std::string const& expectedName, std::string_view currentNamespace, TypeRef const& typeRef)
 {
-    auto const& ref = std::find_if(type.begin(), type.end(), [name, namespace_name](auto&& t)
+    std::string combineName("");
+    if (currentNamespace != typeRef.TypeNamespace())
     {
-        return t.TypeName() == name && t.TypeNamespace() == namespace_name;
-    });
-    REQUIRE(ref.TypeName() == name);
-    REQUIRE(ref.TypeNamespace() == namespace_name);
-    return ref;
+        combineName += typeRef.TypeNamespace();
+        combineName += ".";
+    }
+    combineName += typeRef.TypeName();
+    REQUIRE(combineName == expectedName);
+}
+
+void VerifyReturnType(ElementType const& expectedType, std::string const& expectedTypeRef, MethodDef const& method, TypeDef const& parent)
+{
+    auto const& methodSig = method.Signature();
+    REQUIRE(method.ParamList().first[0].Name() == "returnVal");
+    switch (expectedType)
+    {
+    case ElementType::Void:
+        REQUIRE(!methodSig.ReturnType());
+        break;
+    case ElementType::End:
```

Comment:
This method seems a bit clunky. Why is ElementType::End involved? Why is expectedTypeRef only checked in the ElementType::End case? I fear this won't really scale, especially when we start handling generic types.

Reply:
Will fix later.  This works with currently supported types.  Opened a bug to refactor this to more easily support generics/nested types in the future.